core/lib/src/request/request.rs

```diff
@@ -490,7 +490,7 @@ impl<'r> Request<'r> {
             // where the client accepts the thing at all?
             self.accept()
                 .map(|accept| accept.preferred().media_type())
-                .or(Some(&ANY))
+                .or_else(|| Some(&ANY))
```

Comment:
This seems unnecessary. Constructing `Some(&ANY)` is free, so this just adds syntactic noise.